plugins/woocommerce-blocks/tests/e2e/tests/product-collection/product-collection.block_theme.spec.ts

```diff
@@ -1590,6 +1590,62 @@ test.describe( 'Product Collection', () => {
 			await expect( products ).toHaveText( expectedProducts );
 		} );
 	} );
+
+	test.describe( 'Extensibility - JS events', () => {
+		test( 'emits event about block rendered', async ( {
+			pageObject,
+			page,
+		} ) => {
+			await pageObject.createNewPostAndInsertBlock();
+
+			await page.addInitScript( () => {
+				let eventFired = 0;
+				window.document.addEventListener(
+					'wc-blocks_product_list_rendered',
+					() => {
+						window.eventFired = ++eventFired;
+					}
+				);
+			} );
+
+			await pageObject.publishAndGoToFrontend();
+			await page.waitForLoadState();
+
+			const eventFired = await page.evaluate( async () => {
+				return window.eventFired;
+			} );
+
+			expect( eventFired ).toBe( 1 );
```

Comment:
For stability, I'd recommend using `expect.poll` here so that we don't have to use `waitForLoadState` which is not very reliable:

```js
await pageObject.publishAndGoToFrontend();
await expect.poll( page.evaluate( 'window.eventFired' ) ).toBe( 1 );
```